src/transaction/TransactionController.test.ts

```diff
@@ -590,20 +590,20 @@ describe('TransactionController', () => {
     await expect(result).rejects.toThrow('foo');
   });
 
-  it('should fail transaction if gas calculation fails', async () => {
+  it('should not fail transaction if gas calculation fails', async () => {
```

Comment:
Why are we negating the test case here?

Reply:
Thanks for the review!

Now if the gas calculation fails the transaction should not fail
Instead of failing, it will conduct a variable that contains the error

Reply:
There doesn't seem to be anything that would cause the `estimateGas` query to throw now. Do we still need `mockFlags.estimateGas = 'Uh oh'`? Additionally, should we test that the return object contains `estimateGasError` in that case?

Reply:
Also, according to the code, there are three cases in which the return object could include `estimateGasError`: 

* If `gasBN.gt(maxGasBN)`
* If `isCustomNetwork`
* If `paddedGasBN.lt(maxGasBN)`

Should we verify that these cases are adequately tested?

Reply:
I updated with two test cases with the estimateGasError replacing the older one: 
* For custom Network
* For Mainnet

The gasBN and paddedGasBN originated by gasHex and gasHex it would have a value if the rpc method eth_estimateGas did not fail.

Please let me know if there is something that I could improve on those new test cases or if I'm missing some test case that I'm not visualizing right now

Reply:
Let's put these hands working ahah! 
Thank you a lot @mcmire, for helping with this review with showing what's needed!

Reply:
Updated the PR but the second test case:
* gasBN is greater than maxGasBN, and the query to eth_estimateGas does fail -> needs to succeed because gasBN it's calculated by the gasHex that appends its value from eth_estimateGas